bundles/core/org.eclipse.smarthome.core.audio/src/main/java/org/eclipse/smarthome/core/audio/URLAudioStream.java

```diff
@@ -52,27 +57,32 @@ public URLAudioStream(String url) throws AudioException {
 
     private InputStream createInputStream() throws AudioException {
         try {
-            if (url.toLowerCase().endsWith(".m3u")) {
-                InputStream is = new URL(url).openStream();
-                String urls = IOUtils.toString(is);
-                for (String line : urls.split("\n")) {
-                    if (!line.isEmpty() && !line.startsWith("#")) {
-                        url = line;
-                        break;
-                    }
-                }
-            } else if (url.toLowerCase().endsWith(".pls")) {
-                InputStream is = new URL(url).openStream();
-                String urls = IOUtils.toString(is);
-                for (String line : urls.split("\n")) {
-                    if (!line.isEmpty() && line.startsWith("File")) {
-                        Matcher matcher = plsStreamPattern.matcher(line);
-                        if (matcher.find()) {
-                            url = matcher.group(1);
+            final String filename = url.toLowerCase();
+            final String extension = AudioStreamUtils.getExtension(filename);
+            switch (extension) {
+                case M3U_EXTENSION:
+                    final InputStream isM3U = new URL(url).openStream();
+                    for (final String line : IOUtils.readLines(isM3U)) {
+                        if (!line.isEmpty() && !line.startsWith("#")) {
+                            url = line;
                             break;
                         }
                     }
-                }
+                    break;
+                case PLS_EXTENSION:
+                    final InputStream isPLS = new URL(url).openStream();
```

Comment:
Good point. I changed it. Maybe the `try {} catch {}` block is placed incorrect now and should be refactored / split up as well.